riptide-core/src/main/java/org/zalando/riptide/Requester.java

```diff
@@ -118,7 +133,11 @@ public final Requester contentType(final MediaType contentType) {
         private CompletableFuture<ClientHttpResponse> send() throws IOException {
             final AsyncClientHttpRequest request = createRequest();
             worker.write(request, entity);
-            return new ListenableToCompletableFutureAdapter<>(request.executeAsync());
+            final ListenableFuture<ClientHttpResponse> original = request.executeAsync();
+
+            final CompletableFuture<ClientHttpResponse> future = preserveCancelability(original);
+            original.addCallback(future::complete, future::completeExceptionally);
```

Comment:
It's similar, but `original` is a `ListenableFuture` and it doesn't support `whenComplete`.

Reply:
To bad, I see a great potential of simplification by using `whenComplete(forwardTo())`. At least for me it is much easier to understand how forwarding of results works together with cancelling. Especially, if it would be integrated into `preserveCancelabiliy()`, this would maker absolutely clear how the forwarding works and ensure that it is done properly.

Reply:
I have usages in the code bases that only need `preserveCancelability` as well as ones that only use `forwardTo`. I hope this will become a lot simpler if we can leverage Java 9 to its full extent.